src/python_testing/TC_MCORE_FS_1_4.py

```diff
@@ -0,0 +1,485 @@
+
+    def setup_class(self):
+        super().setup_class()
+
+        # Get the path to the TH_FSA (fabric-admin and fabric-bridge) app from the user params.
+        th_fsa_app_path = self.user_params.get("th_fsa_app_path")
+        if not th_fsa_app_path:
+            asserts.fail("This test requires a TH_FSA app. Specify app path with --string-arg th_fsa_app_path:<path_to_app>")
+        if not os.path.exists(th_fsa_app_path):
+            asserts.fail(f"The path {th_fsa_app_path} does not exist")
+        th_fsa_admin_path = self.user_params.get("th_fsa_admin_path")
+        if not th_fsa_admin_path:
+            asserts.fail("This test requires a TH_FSA_ADMIN app. Specify app path with --string-arg th_fsa_admin_path:<path_to_app>")
+        if not os.path.exists(th_fsa_admin_path):
+            asserts.fail(f"The path {th_fsa_admin_path} does not exist")
+        th_fsa_bridge_path = self.user_params.get("th_fsa_bridge_path")
+        if not th_fsa_bridge_path:
+            asserts.fail("This test requires a TH_FSA_BRIDGE app. Specify app path with --string-arg th_fsa_bridge_path:<path_to_app>")
+        if not os.path.exists(th_fsa_bridge_path):
+            asserts.fail(f"The path {th_fsa_bridge_path} does not exist")
+
+        # Get the path to the TH_SERVER_NO_UID app from the user params.
+        th_server_app = self.user_params.get("th_server_no_uid_app_path", None)
+        if not th_server_app:
+            asserts.fail("This test requires a TH_SERVER_NO_UID app. Specify app path with --string-arg th_server_no_uid_app_path:<path_to_app>")
+        if not os.path.exists(th_server_app):
+            asserts.fail(f"The path {th_server_app} does not exist")
+
+        # Create a temporary storage directory for keeping KVS files.
+        self.storage = tempfile.TemporaryDirectory(prefix=self.__class__.__name__)
+        logging.info("Temporary storage directory: %s", self.storage.name)
+
+        self.th_fsa_bridge_address = "::1"
+        self.th_fsa_bridge_port = 5543
+        self.th_fsa_bridge_discriminator = random.randint(0, 4095)
+        self.th_fsa_bridge_passcode = 20202021
+
+        self.th_fsa_controller = FabricSyncApp(
+            th_fsa_app_path,
+            th_fsa_admin_path,
+            th_fsa_bridge_path,
+            storage_dir=self.storage.name,
+            paa_trust_store_path=self.matter_test_config.paa_trust_store_path,
+            bridge_port=self.th_fsa_bridge_port,
+            bridge_discriminator=self.th_fsa_bridge_discriminator,
+            bridge_passcode=self.th_fsa_bridge_passcode,
+            vendor_id=0xFFF1)
+
+        # Get the named pipe path for the DUT_FSA app input from the user params.
+        dut_fsa_stdin_pipe = self.user_params.get("dut_fsa_stdin_pipe", None)
+        if dut_fsa_stdin_pipe is not None:
+            self.dut_fsa_stdin = open(dut_fsa_stdin_pipe, "w")
+
+        self.th_server_port = 5544
+        self.th_server_discriminator = random.randint(0, 4095)
```

Comment:
nit, we have a 1 in 4096 chance of a collision where we might have an issue during discovery. We could just do 

`self.th_fsa_bridge_discriminator = random.randint(0, 4094)` with a comment saying we add one for the th_server so we intentionally made max one less. Then here we add 1 to the previous result to prevent collision.

Reply:
Good point!